avm/res/network/load-balancer/main.bicep

```
@@ -91,6 +91,13 @@ var frontendIPConfigurationsVar = [
           }
         : null
     }
+    zones: contains(frontendIPConfiguration, 'subnetId') && !empty(frontendIPConfiguration.subnetId)
+    ? [
+        '1'
+        '2'
+        '3'
+      ]
+    :null
```

I wonder if it should be possible for users to also set explicit zones if needed. Like

```suggestion
    zones: contains(frontendIPConfiguration, 'zones') ? map(frontendIPConfiguration.zones, zone => string(zone)) : !empty(frontendIPConfiguration.?subnetResourceId)
    ? [
        '1'
        '2'
        '3'
      ]
    : null
```
Where one could set an int array of zones, and if not and a subnetResourceId was provided, it would fallback to the default

---

On that note - it should also check for `subnetResourceId` not `subnetId` in line 80

---

The zonal check looks good - please just also update the `subnetId` item